### PR TITLE
Ft crud operations buses

### PR DESCRIPTION
**Pull request by mugiisha**
**What does this PR do?**

Create, Read, Update and Delete buses

**Description of Task to be completed?**

Integration of buses using redux
Created reducers, store, and actions for CRUD operations for buses.

**How should this be manually tested?**

step 1: clone git repo if not yet done.
step 2: cd Avengers-Phantom-FE.
step 3: yarn or npm install to install dependencies.
step 5: Use google chrome for better rendering (google chrome redux).

**APIS Used**

GET ---api/v1/buses for getting all buses
GET ---api/v1/buses/:uuid for getting bus details.
POST ---api/v1/buses/ for creating buses
PATCH --api/v1/buses/:uuid for updating bus

**Pages to navigate to**

GET all buses-----URL/dashboard/buses
GET single bus-----URL/dashboard/bus?id=:busid
Register bus -----URL/dashboard/bus/register
update bus----URL/dashboard/bus/update?id=:busid

### CODE REVIEW

**Review by Anguandia — APPROVED** [review]
Hello @mugiisha, good work please use this [markdown](https://www.markdownguide.org/basic-syntax/) guide in subsequent PR descriptions. The headings especially should standout from the content

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good job @mugiisha
- Please complete the _how to test this PR_ section with details of which page to navigate to
- what actions to execute once on the page
- what responses to expect
- screenshots of the process would be a great enhancement
- also, improve the styling of your PR description using this [markdown](https://www.markdownguide.org/basic-syntax/) guide. The headings especially should standout from their content/body

File: src/redux/Action/fetchallbuses.js [hunk 1]
```diff
@@ -1,14 +1,101 @@
...
+  dispatch({
+    type: actions.FETCH_ALL_BUSES,
+    payload: res.data.buses.rows,
+  });
+
+  };
+
+  export const deletebus = (id) => async (dispatch) => {
+    try{
+      const res = await axios.delete(`https://new-avengers-be-deploy.herokuapp.com/api/v1/buses/${id}`,{
```
Comment (Anguandia) [inline src/redux/Action/fetchallbuses.js]: Please, make it a point to read the backend url dynamically from the environment. I want to test with my local backend

**Review by Anguandia — CHANGES_REQUESTED** [review]

**Review by Anguandia — APPROVED** [review]

File: src/Components/dashboard/buses/Validations.jsx [hunk 2]
```diff
@@ -34,8 +34,8 @@ export const FormValidation = ()=>{
         if("type" in fieldValues){
             tmp.type = fieldValues.type?"":"Required field."
         }
-        if("platenumber" in fieldValues){
-            tmp.platenumber = fieldValues.type? "":"Required field."
+        if("plateNumber" in fieldValues){
+            tmp.plateNumber = fieldValues.type? "":"Required field."
             // if (fieldValues.platenumber){
```
Comment (Byadiso) [inline src/Components/dashboard/buses/Validations.jsx]: try to remove these commented lines, if they are not needed

File: src/Components/signin/Signin.jsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ const theme = createTheme({
     },
   },
 });
+const baseUrl = process.env.BACKEND_URL
```
Comment (Byadiso) [inline src/Components/signin/Signin.jsx:33]: Is this working ? without dotenv configurated?

Comment (mugiisha) [reply src/Components/signin/Signin.jsx:33]: dotenv is configurated in server.js

File: src/Components/dashboard/buses/BusDetail.jsx [hunk 4]
```diff
@@ -5,8 +5,26 @@ import Photo from "../../../static/images/busbackground.jpg";
 import { Link } from "react-router-dom";
 import AddAPhotoIcon from "@mui/icons-material/AddAPhoto";
 import DashboardLayout from "../../../Layouts/Dashboard";
+import Sidebar from "../sidebar/Sidebar.jsx";
+import DashNavbar from "../dashnavbar/DashNavBar.jsx";
```
Comment (abraham-kwizera) [inline src/Components/dashboard/buses/BusDetail.jsx]: As we have a new integrated dashboard design, I hope that we don't need this `DashNavbar` still.

File: src/Components/dashboard/buses/BusDetail.jsx [hunk 5]
```diff
@@ -5,8 +5,26 @@ import Photo from "../../../static/images/busbackground.jpg";
...
+    const getBus =()=>{
+      dispatch(getBusById(busId))
+    }
+    
+    useEffect(() => {
+      getBus()
+    },[])
+    
+    const bus = useSelector(state => state.fetchbuses?.busById)
+    console.log(bus)
```
Comment (abraham-kwizera) [inline src/Components/dashboard/buses/BusDetail.jsx]: Let's remove these console log messages in this file.

File: src/Components/dashboard/buses/updateBus.jsx [hunk 6]
```diff
@@ -5,13 +5,47 @@ import AddAPhotoIcon from "@mui/icons-material/AddAPhoto";
...
+      dispatch(getBusById(busId))
+    }
+    
+    useEffect(() => {
+      getBus()
+    },[])
+  
+  
+  const bus = useSelector(state => state?.fetchbuses?.busById)
+  console.log(bus)
```
Comment (abraham-kwizera) [inline src/Components/dashboard/buses/updateBus.jsx]: Lets remove the console logs messages in this file

File: src/redux/Action/fetchallbuses.js [hunk 7]
```diff
@@ -1,14 +1,102 @@
...
+        }
+      });
+      const message = res.data.message
+    dispatch({
+      type: actions.DELETE_BUS,
+      payload: {id:id,message:res.data.message},
+    });
+    notify(message)
+    }catch(err){
+      console.log(err.response.data)
```
Comment (abraham-kwizera) [inline src/redux/Action/fetchallbuses.js:37]: As long as you have implemented a toast notification pop up, hope you have remove these console log message in this file.

**Review by abraham-kwizera — COMMENTED** [review]
Hey @mugiisha you did greate, but i did left some comment reviews of suggested changes in some file. Let's work on it before we go 🚀

**Review by Byadiso — APPROVED** [review]
Just squash these commits to 2 or 1 commit if possible!

**Review by abraham-kwizera — APPROVED** [review]
@mugiisha just [Resolve Conflicts](https://github.com/atlp-rwanda/Avengers-Phantom-FE/pull/44#:~:text=package%2Dlock.json,webpack.config.js) and we go 🚀

**Pull request merged**